webapp/src/pages/ResponsePage/ResponsePage.js

```diff
@@ -1,91 +1,233 @@
 import './ResponsePage.css'
+import { withTranslation } from 'react-i18next';
+import TagModal from './components/TagModal'
+import DeleteModal from './components/DeleteModal'
+import { eventService } from '../../services/events/events.service'
+import { tagResponse } from '../../services/responseTag/responseTag.service'
 import { applicationFormService } from '../../services/applicationForm/applicationForm.service'
 import { fetchResponse } from '../../services/responsePage/responsePage.service'
+import { tagList } from '../../services/tagList/tagList.service'
+
 
 class ResponsePage extends Component {
     constructor(props) {
         super(props);
         this.state = {
-
+            tagMenu: false,
+            error: false,
+            addNewTags: [],
+            eventLanguages: []
         }
     };
 
-
     componentDidMount() {
-        this.fetchForm()
-        this.fetchData()
+        this.fetchForm();
+        this.fetchData();
+        this.fetchTags();
+        this.fetchEvent();
     }
 
+    // Data Functions
+    // Fetch Event Details
+    fetchEvent() {
+        eventService.getEvent().then(response => {
+            this.setState({
+                eventLanguages: ["en", "fr"]
+            })
+        });
+    }
+
+
     // Fetch Form
     fetchForm() {
         applicationFormService.getForEvent(this.props.event.id).then(response => {
             this.setState({
                 applicationForm: response.formSpec
             })
-        })
-
+        });
     }
 
 
     // Fetch Data
     fetchData() {
         let params = {
             id: this.props.match.params.id
-        }
+        };
         fetchResponse(params).then(response => {
             this.setState({
                 applicationData: response
             })
-        })
-
-
+        });
     }
 
 
+    // Misc FUnctions
     // Go Back
     goBack() {
         this.props.history.goBack();
     }
 
 
+    error(error) {
+        this.setState({
+            error: error
+        })
+    }
+
+
+    // Tag Functions
+    // Populate tag list
+    fetchTags() {
+        tagList.list().then(response => {
+            this.setState({
+                tagList: response
+            })
+        });
+    }
+
+
+    // Toggle Tag Menu
+    toggleTags(list) {
+        this.setState({
+            tagMenu: list ? false : true
+        })
+    }
+
+
+    // Post Tag
+    postTag(tag, type) {
+        type == "responseTag" ? this.postResponseTag(tag) : this.postTagList(tag);
+    }
+
+
+
+    // Post Response API
+    postResponseTag(tag) {
+        const updateApplicationData = this.state.applicationData;
+        const updateTagList = this.state.tagList;
+
+        tagResponse.post(tag).then(response => {
+            if (response.status == 201) {
+                let getTag = this.state.tagList.filter(tag => {
+                    if (tag.id == response.tag_id) {
+                        return tag
+                    }
+                });
```

Comment:
FYI: Since you're just looking for a single tag, you can do this:
`const getTag = this.state.tagList.find(tag => tag.id === response.tag_id);`
(the result will be a single tag, rather than an array of tags)

Reply:
Thanks !